### DISCUSSION TITLE
Unsubscribing notifications

### DISCUSSION THREAD

**Discussion by loxy** [taiga-family/taiga-ui#952] [Q&A] [answered]
Hi!

Notifications are implemented by Observables. Cool. Just to confirm: Unsubscribing from a Notification is only needed to cancel a subscription and prevent the Notification from opening, right? There is no case where a Notification leaks memory because it always completes (at least if the user clicks on close), right?

Best regards,
Kersten

**Comment by waterplea** [comment, marked as answer]
Completion = closed notification. You can unsubscribe to close it programatically and if user closes it or it autocloses by timeout — Observable completes. So yes, memory leaks are impossible.

**Reply by loxy** [reply]
Thank you! Can I add this to the docs? So other developers don't need to scan the source code. There are also some small typos in the current text:

> You can also just ***unsubcribe*** from a stream to ***hode*** a notification (this observable is returned from open method). You can save a subscription for that or use takeUntil like tools from RxJs

**Reply by waterplea** [reply]
Sure, go ahead and drop a PR!
